### ISSUE TITLE
[Accessibility] Indentation resized view on already-space-indented code

### ISSUE THREAD

**Issue by m13253** [microsoft/vscode#187116]
### The original Reddit post
I recently came across a 4-year-old Reddit post: [Nobody talks about the real reason to use Tabs over Spaces](https://www.reddit.com/r/javascript/comments/c8drjo/nobody_talks_about_the_real_reason_to_use_tabs/)

This post can be summarized as:
1. The original poster was a space-indentation user and moved to a new company.
2. The original poster wanted to enforce the 4-space indentation rule in a new repo.
3. Two coworkers, **both visually impaired,** contacted the poster, saying they will have to convert in-and-out for each commit if the new rule was to be enforced, because:
   * One of them had to use an extremely large font on a normal monitor, requiring 1-column tabs for indentation.
   * The other, instead, used an ultra-wide monitor, consequently requires 8-column tabs for indentation.
4. The original poster was convinced to use tab for this a11y reason, and wrote that story.

### My interview with fellow developers

The topic of using tab or space is usually very sensitive to most developers. Therefore, before I came here, I interviewed 5 developers around me, plus myself, and gathered some opinions below.

The opinions are:
1. All space users except for me, refuse to consider switching to tab for this reason.
2. We all agree that converting existing codebases to tab is never a possible solution.
3. Automatically converting spaces to tab is destructive and lossy. (Converting back may not produce the same results.)
4. All of us except one thinks **code editors should consider adding an indentation resized view** in their a11y configuration page.
5. Reading code on small-screen devices might be another valid reason to allow indentation resized view.
6. ~~One of my friend says, “rendering my code in anything other than 4 spaces is an offense of my rights. And I would be anything but a developer if I get old and require visual-aid.”~~ —He is no longer my friend now.

### My proposal

I propose an a11y feature that can edit a file written in X-space indentation in Y-column indentation. I call it “indentation resized view”.

This feature should:
1. Be lossless and never modify the file itself.
2. UI can distinguish between the detected indentation and normal spaces. (Especially when X == 1 or Y == 1.)
3. Properly handle the case when the user presses a key near a detected indentation, which may increase or decrease the indentation level, or break the indentation into two parts.

### Disclaimers
1. I’m **not** visually impaired. But a quick search here and on the extension marketspace shows no related info. I hope to see more responses from those who are. **Let the people in need decide whether this is really a feature to request.**
2. I have been writing code using spaces. Please refrain from considering me trying to use the excuse of a11y to “evangelize” tabs.

**Comment by rperez030** [comment]
In theory, such feature could be beneficial for magnification users. It wouldn't solve the problem for screen reader users though. a character space is a character space and will always take a cell in a Braille display. Here I would be concerned about the keyboard behavior. As a screen reader user, I wouldn't expect the cursor to jump over blank spaces if I'm moving character by character. That could be particularly confusing if I'm using speech only. I definitely think the possibility should be explored.

**Comment by qwikag** [comment]
> All space users except for me, refuse to consider switching to tab for this reason.
For what reason?

There is technically no reason to stay on spaces other than you are not smart enough to change or your TAB key is busted.

**Comment by qwikag** [comment]
I would propose a different solution:
VSCode enforces tabs, and not have an option of space indentation.

**Comment by dsherret** [comment]
Probably duplicate of https://github.com/microsoft/vscode/issues/132776
